### ISSUE TITLE
Select component lacks focus indicator in Windows High Contrast Mode 

### ISSUE THREAD

**Issue by kinueng** [carbon-design-system/carbon#7301]
## Environment

> Operating system

Windows 10

> Browser

Firefox 83.0

> Assistive technology used to verify

Windows High Contrast Mode

## Detailed description

> What version of the Carbon Design System are you using?

I used the website to test, Carbon Components React v7.24.0 at the time of this writing.
https://react.carbondesignsystem.com/?path=/story/select--default

Also tested on Carbon Components
https://www.carbondesignsystem.com/components/select/usage/#main-content

> What did you expect to happen?

I expected to see some kind of focus indicator when using the keyboard to tab the focus onto the Select component.  A related issue https://github.com/carbon-design-system/carbon/issues/6882 was closed as fixed by a PR, but I am still seeing the issue with Select's focus indicator on Windows 10 High Contrast Mode.

> What happened instead?

There was no visual focus indicator on the Select component.  I confirm the focus is on the Select by pressing the Spacebar to open the Select's dropdown menu.

> What WCAG 2.1 checkpoint does the issue violate?

https://www.ibm.com/able/guidelines/ci162/focus_visible.html

**Comment by dakahn** [comment]
Seems specific to Firefox. Working as expected in Edge/Chrome
![image](https://user-images.githubusercontent.com/40970507/99564157-b80f3a80-298f-11eb-8a2c-9fcbe695983f.png)

**Comment by dakahn** [comment]
Firefox isn't picking up our system color settings -- though it is weirdly adhering to other styles
![image](https://user-images.githubusercontent.com/40970507/99569231-fc9dd480-2995-11eb-8d60-729072e29172.png)

**Comment by kinueng** [comment]
Hi @dakahn and @tw15egan,

Re: https://github.com/carbon-design-system/carbon/issues/7301#issuecomment-729856889

After I opened this issue, I noticed in the fix from https://github.com/carbon-design-system/carbon/pull/7036 requires changes to the default Firefox settings in `about:config`.

> For Firefox, you'll need to open it up and type about:config into the address bar. Then, search for layout.css.prefers-contrast and set it to true.

**I retested by:**
1.  Set `layout.css.prefers-contrast` to `true` in Firefox 83.0 `about:config`
2. Restarted Firefox
3. Retest by tabbing to the Select on https://react.carbondesignsystem.com/?path=/story/select--default
4. Observed the white border on the Select becomes thicker when in focus

**Questions:**
1.  Is a thicker border with the same color sufficient to satisfy [IBM's Checkpoint 2.4.7 Focus Visible Rationale](https://www.ibm.com/able/guidelines/ci162/focus_visible.html) that `All focus indicators should be highly visible and distinguishable.`?
2.  Is there a way to avoid changing the default settings in Firefox to achieve High Contrast Mode support?

I know there is some recent activity in another issue I opened for the UI Shell, https://github.com/carbon-design-system/carbon/issues/7313#issuecomment-729889137

**Comment by tw15egan** [comment]
@kinueng Currently, that is the only way to enable HCM mode in Firefox. It is a newer experimental feature, so we're quickly trying to bring support. I would imagine once it is properly supported / out of experimental, they will add it in their normal options and you won't have to dig in to turn it on. 

I'm updating the focus styles to use the `dotted` property, so it will be a much bigger visual change than just a 1px increase in border width. 

I'll have a PR up shortly and I'll tag you in it so you can help verify the fixes 👍

**Comment by kinueng** [comment]
@tw15egan If setting `layout.css.prefers-contrast` to `true` in Firefox `about:config` is a required step to support Win10 HCM with Firefox, will the Carbon accessibility documentation be updated with details that Win10 HCM on Firefox requires a latest Firefox version and a non-default setting?

Here is the current accessibility documentation for the `<Select>` component https://www.carbondesignsystem.com/components/select/accessibility/
